core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -169,14 +174,24 @@ private[spark] object ReliableCheckpointRDD extends Logging {
     val bufferSize = env.conf.getInt("spark.buffer.size", 65536)
 
     val fileOutputStream = if (blockSize < 0) {
-      fs.create(tempOutputPath, false, bufferSize)
+      val checkpointCodec = env.conf.get("spark.checkpoint.compress.codec", "none")
+      val fileStream = fs.create(tempOutputPath, false, bufferSize)
+      if (CompressionCodec.ALL_COMPRESSION_CODECS_SHORT.contains(checkpointCodec)) {
+        val compressionCodec = CompressionCodec.createCodec(env.conf, checkpointCodec)
+        logInfo(s"Compressing using $checkpointCodec.")
+        compressionCodec.compressedOutputStream(fileStream)
+      } else {
+        fileStream
```

Comment:
This repeated pattern can be rewritten as indicated above https://github.com/apache/spark/pull/17024/files#r102418860